src/write.jl

```diff
@@ -185,6 +188,17 @@ function writerow(buf, pos, len, io, sch, row, cols, opts)
     return
 end
 
+# Special treatment for `nothing`. It can be treated like `missing`
+# if the `treatnothingasmissing` option is set to `true`.  Otherwise,
+# an empty string is written.
+function writecell(buf, pos, len, io, ::Nothing, opts)
+    if opts.treatnothingasmissing
+        writecell(buf, pos, len, io, missing, opts)
+    else
+        writecell(buf, pos, len, io, "", opts)
```

Comment:
I'm not comfortable with this; I think this branch should throw an informative error, pointing the user to `treatnothingasmissing=true` or using `Tables.transform`. The point of `nothing` in the language is to be a non-useful "null" value that is dealt with as quickly as possible; I think it's an anti-pattern to start giving it behavior.

Reply:
Fair enough :-)

Reply:
@quinnj I have updated code to throw error when `nothing` is encountered.  I also added the column index to the `writecell` function everywhere so we can give the user a little more clue about where the `nothing` was found.  I was thinking about adding the row number too if we like this approach.

@nalimilan I did a separate implementation to allow passing in a cell-based transformer function.  That works too but my performance test shows 2x regression for a 100000x3 data frame of Float64's.  I don't know how to optimize it.  You can find the code at my fork https://github.com/tk3369/CSV.jl/tree/cell-based-transform

Here's the perf test harness:
```
$ cat test/perf_write.jl 
using DataFrames, CSV, BenchmarkTools

function test(n)
    df = DataFrame(x = rand(n), y = rand(n), z = rand(n))
    @benchmark CSV.write("/tmp/tom.csv", $df)
end

for i in 4:7
    rows = 10^i
    println(rows, " rows: ", test(rows))
end
```

Performance result from this PR:
```
10000 rows: Trial(3.081 ms)
100000 rows: Trial(41.997 ms)
1000000 rows: Trial(415.127 ms)
10000000 rows: Trial(5.391 s)
```

Performance result from `cell-based-transform` branch:
```
10000 rows: Trial(6.696 ms)
100000 rows: Trial(78.961 ms)
1000000 rows: Trial(779.919 ms)
10000000 rows: Trial(10.350 s)
```

Performance from version 0.5.22:
```
10000 rows: Trial(2.589 ms)
100000 rows: Trial(40.224 ms)
1000000 rows: Trial(418.302 ms)
10000000 rows: Trial(5.852 s)
```

Reply:
You have to make `transform::F` with `F<:Union{Function, Nothing}` in the definition of `Options` to avoid dynamic dispatch.

Also you could simplify the code a bit by using `(col, val) -> val` instead of `nothing` as the default: the compiler will probably be able to optimize this out.

Reply:
Submitted PR #564. Your suggestion solved the performance issue.  

We can close this PR after you guys review the other one and agree that is the right direction.